client/core/core.go

```diff
@@ -6173,6 +6243,39 @@ func (c *Core) removeWaiter(id string) {
 	c.waiterMtx.Unlock()
 }
 
+// peerChange is called by a wallet backend when the peer count changes or
+// cannot be determined. A wallet state note is always emitted. In addition to
+// recording the number of peers, if the number of peers is 0, the wallet is
+// flagged as not synced. If the number of peers is >0 and was previously zero,
+// a resync monitor goroutine is launched to keep the wallet as not synced until
+// it's SyncStatus method reports it has fully synchronized with the network.
+// The sync status monitor goroutine will repeatedly emit wallet state notes,
+// and once sync has been restored, a wallet balance note will be emitted.
+func (c *Core) peerChange(w *xcWallet, numPeers uint32) {
+	if numPeers == 0 {
+		c.log.Warnf("Wallet for asset %s has zero network peers!", unbip(w.AssetID))
+	} else {
+		c.log.Debugf("New peer count for asset %s: %v", unbip(w.AssetID), numPeers)
+	}
+
+	w.mtx.Lock()
+	wasDisconnected := w.peerCount == 0
+	w.peerCount = numPeers
+	if numPeers == 0 {
```

Comment:
Ah, one of the biggest goals of this work was to emit a warn/error notification when peers changes from >0 to 0.  TODO!

We can even update `startWalletSyncMonitor` to show an info ntfn in the reverse case, when sync is reestablished.